shared/constants/push.js

```diff
@@ -1,17 +1,74 @@
 // @flow
+import * as I from 'immutable'
 import * as Types from './types/push'
+import * as ChatConstants from './chat2'
+import * as ChatTypes from './types/chat2'
+import {isIOS} from '../constants/platform'
+import {isDevApplePushToken} from '../local-debug'
+import logger from '../logger'
 
-export const tokenTypeApple: Types.TokenType = 'apple'
-export const tokenTypeAppleDev: Types.TokenType = 'appledev'
-export const tokenTypeAndroidPlay: Types.TokenType = 'androidplay'
-
+export const tokenType = isIOS ? (isDevApplePushToken ? 'appledev' : 'apple') : 'androidplay'
 export const androidSenderID = '9603251415'
-
 export const permissionsRequestingWaitingKey = 'push:permissionsRequesting'
 
-export const initialState: Types.State = {
-  permissionsPrompt: false,
-  tokenType: null,
-  token: '',
+export const makeInitialState: I.RecordFactory<Types._State> = I.Record({
   hasPermissions: true,
+  showPushPrompt: false,
+  token: '',
+  tokenType: null,
+})
+
+export const normalizePush = (n: any): ?Types.PushNotification => {
+  try {
```

Comment:
why the try/catch?

Reply:
because this is very untrusted input, comes basically raw from push and i never want this to crash us out